zilliqa/src/consensus.rs

```diff
@@ -1053,6 +1053,16 @@ impl Consensus {
 
         let node_config = &self.config;
 
+        if self.block_is_first_in_epoch(parent.number() + 1) {
+            // Here we ignore any errors and proceed with block creation, under the philosophy that
+            // it is better to have the network run but with broken epochs, than it would be to
+            // crash the entire network on a failing epoch transition.
```

Comment:
Our consensus now depends on successful EVM calls - and not just read-only checks as before, but a complex write transaction.

This raises the question of what to do if the `tick_epoch()` call fails for any reason. Should we assume that it must never fail, and crash the node if it does (potentially crashing the network if a buggy state is reached)? The approach taken here is to soft-ignore errors; the network then will remain alive, but with a static validator set since epoch changes will be broken. Measures can then be taken to upgrade to fix the root cause of the issue (e.g. upgrading the deposit contract) to restore normal functionality.

But, especially right now in the early testnet phases, perhaps hard-crashing the network on any bugs might be preferable to discover any issues immediately.